### ISSUE TITLE
feature request: support eliom file extensions

### ISSUE THREAD

**Issue by jrochel** [ocaml/ocaml-lsp#353]
This would be ".eliom" for OCaml and ".eliomi" for Ocaml interface.
The patch is trivial, but I still can't seem to get it to work.
I hope someone who knows this source code will be able to implement this within 5 seconds.

**Comment by jordanrmerrick** [comment]
Do you mean you want to apply OCaml syntax to `.eliom` and `.eliomi` files? If so, a current workaround can be found [here](https://discuss.ocaml.org/t/why-eliom-has-its-own-file-name-extensions-eliom/1717).

**Comment by jrochel** [comment]
Indeed, that's what I want and what I'm already doing. However, the recognition of whether the current file is an interface file or and implementation file is done in ocaml-lsp-server, if I'm not mistaken.

I did add the file extensions ".eliom" and ".eliomi" to `ocaml-lsp-server/src/document.ml` and to `ocaml-lsp-server/src/inference.ml`, but that didn't seem to do the trick.

**Comment by jrochel** [comment]
https://github.com/ocaml/ocaml-lsp/pull/354

**Comment by jrochel** [comment]
The PR has been merged, but I still cannot get the recognition of .eliomi files to work. 

> Even when sending "languageId": "ocaml.interface" my .eliomi files are still treated as Impl files. When I edit some .mli file the file is correctly treated as an Intf file, while in the logs I see no occurrence of ocaml.interface, but only ever "languageId":"ocaml". Also, in the ocaml-lsp source code the mapping from "ocaml.interface" is to OCaml and not to (OCaml, Intf). I suspect that there is some magic somewhere that uses the file extension of the path to determine whether the file is an Intf or an Impl, but I don't understand why it only works for .ml/.mli but not .eliom/.eliomi.
